classes/CSteamSharedfile.js

```diff
@@ -0,0 +1,223 @@
+				}
+
+				// Make callback when ID was resolved as otherwise owner will always be null
+				callback(null, new CSteamSharedfile(this, sharedfile));
+			});
+
+		} catch (err) {
+			callback(err, null);
+		}
+	}, "steamcommunity");
+};
+
+function CSteamSharedfile(community, data) {
+	this._community = community;
+
+	// Clone all the data we recieved
+	Object.assign(this, data); // TODO: This is cleaner but might break IntelliSense. I'm leaving the block below to be reactivated if necessary
+
+	/* this.id = data.id;
+	this.type = data.type;
+	this.appID = data.appID;
+	this.owner = data.owner;
+	this.fileSize = data.fileSize;
+	this.postDate = data.postDate;
+	this.resolution = data.resolution;
+	this.uniqueVisitorsCount = data.uniqueVisitorsCount;
+	this.favoritesCount = data.favoritesCount;
+	this.upvoteCount = data.upvoteCount; */
+}
+
+/**
+ * Deletes a comment from this sharedfile's comment section
+ * @param {String} cid - ID of the comment to delete
+ * @param {function} callback - Takes only an Error object/null as the first argument
+ */
+CSteamSharedfile.prototype.deleteComment = function(cid, callback) {
+	this._community.deleteSharedfileComment(this.userID, this.id, cid, callback);
+};
+
+/**
+ * Favorites this sharedfile
+ * @param {function} callback - Takes only an Error object/null as the first argument
+ */
+CSteamSharedfile.prototype.favorite = function(callback) {
+	this._community.favoriteSharedfile(this.id, this.appID, callback);
+};
+
+/**
+ * Posts a comment to this sharedfile
+ * @param {String} message - Content of the comment to post
+ * @param {function} callback - Takes only an Error object/null as the first argument
+ */
+CSteamSharedfile.prototype.comment = function(message, callback) {
+	this._community.postSharedfileComment(this.owner, this.id, message, callback);
+};
+
+/**
+ * Subscribes to this sharedfile's comment section. Note: Checkbox on webpage does not update
+ * @param {function} callback - Takes only an Error object/null as the first argument
+ */
+CSteamSharedfile.prototype.subscribe = function(callback) {
+	this._community.subscribeSharedfileComments(this.owner, this.id, callback);
+};
+
+/**
+ * Unfavorites this sharedfile
+ * @param {function} callback - Takes only an Error object/null as the first argument
+ */
+CSteamSharedfile.prototype.unfavorite = function(callback) {
+	this._community.unfavoriteSharedfile(this.id, this.appID, callback);
+};
+
+/**
+ * Unsubscribes from this sharedfile's comment section. Note: Checkbox on webpage does not update
+ * @param {function} callback - Takes only an Error object/null as the first argument
+ */
+CSteamSharedfile.prototype.unsubscribe = function(callback) {
+	this._community.unsubscribeSharedfileComments(this.owner, this.id, callback);
+};
+
+/**
+ * Downvotes this sharedfile
+ * @param {function} callback - Takes only an Error object/null as the first argument
+ */
+CSteamSharedfile.prototype.voteDown = function(callback) {
```

Comment:
I think I'd prefer to leave voteDown and voteUp out of the library. I want to try to maintain at least a tenuous good relationship with Valve, and the potential for abuse here seems too great. I also can't really think of a valid use-case for bot voting.

Reply:
Alright, I removed both functions